Description: Set can-persist property
 OpenConnect can persist and reconnect when the underlying physical
 connection goes away or changes. It does rather depend on NetworkManager
 not screwing up the routes and leaving the VPN gateway routed *through*
 the VPN itself, and it would be kind of nice to accelerate the reconnect
 by sending SIGUSR2 to openconnect to make it do an immediate reconnect
 when a physical link comes back up, instead of waiting for Dead Peer
 Detection. But basically, it should work.
Author: David Woodhouse <dwmw2@infradead.org>
Origin: upstream, https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/commit/f8e386e2c9522cb93a604eab9e534cb1f98ff18c
Last-Update: 2018-03-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/nm-openconnect-service-openconnect-helper.c
+++ b/src/nm-openconnect-service-openconnect-helper.c
@@ -519,6 +519,9 @@
 	else
 		helper_failed (proxy, "VPN Gateway");
 
+	val = g_variant_new_boolean (TRUE);
+	g_variant_builder_add (&builder, "{sv}", NM_VPN_PLUGIN_CAN_PERSIST, val);
+
 	/* Tunnel device */
 	val = str_to_gvariant (getenv ("TUNDEV"), FALSE);
 	if (val)
